### DISCUSSION TITLE
Jinvoo interruttori e controller tapparelle - Tasmota flash

### DISCUSSION THREAD

**Discussion by reiner348** [indomus/forum#925] [Hardware & Firmware]
Ciao a tutti! Sono nuovo del forum, sono finito su indomus in cerca di consigli su come fare a svincolare i miei interruttori dal server cinese che ogni mese si impalla... Ho scoperto tasmota!

E qui vengono i problemi. Prima di smontare tutta casa volevo provare, con un interruttore di riserva che ho, a flashare tasmota e usare home assistant. Il problema è che non riesco a flashare...

Premessa: ho interruttori Jinvoo comprati su amazon, quelli a 3 switch per le luci e quelli dedicati alle tapparelle per... le tapparelle. Non so se posso mettere il link amazon, nel caso si possa lo aggiungo. Montano un chip TYWE3S

Ho provato un po' tutto:
- flashare usando nodemcu come serial converter -> non va
- flashare usando un serial converter dedicato -> non va
- flashare (in entrambi i casi) usando dei pin appoggiati sull'esp -> non va
- flashare saldando dei pin all'esp -> non va
Ricevo sempre l'errore di timeout perchè atom non riesce a comunicare con l'ESP. Per sicurezza ho provato a flashare direttamentee l'ESP a bordo del nodemcu e sono riuscito tranquillamente al primo colpo, quindi tutta la parte software fino a ottenere il .bin dovrebbe essere corretta.

Credo di riuscire sempre a far andare l'ESP in flash mode mettendo a massa il GPI0, perchè si avvia "diverso" e non lo vedo comparire su smartlife (è configurato, quindi se lo avvio senza mettere a massa GPI0 compare regolarmente). Non riesco a capire se c'è un modo tramite atom per essere sicuri al 100% che sia in modalità flash oppure no, e poi perchè va sempre in timeout e non flasha...

Qualcuno riesce a darmi qualche consiglio su cosa potrei tentare per risolvere il problema?

Una informazione che ho recuperato dalla app smartlife:
- FW modulo principale v1.2.3
- FW modulo MCU v1.2.3
Modulo MCU cosa sarebbe? Perchè leggendo in giro ho notato che potrebbe essere l'MCU che impedisce la comunicazione corretta e andrebbe messo a massa pure il suo RST, però se c'è non saprei individuarlo sulla scheda...
Edit: Proseguendo gli studi ho capito cos'è l'MCU e qual'è. E' il chip nero a 8pin che si vede nella foto un po' a sinistra del TYWE3S. Sono riuscito a fatica a leggere cosa c'è scritto, è un 83a04a-3, chip che serve per gli i/o touch. Da come è messo, direi che legge i 3 input a sfioro e li manda, non so di preciso come perchè le piste scompaiono, al TYWE3S. Forse è lui a impedire il flash... Non ho trovato un pin RST per tenerlo spento, potrei provare a tagliargli il pin di alimentazione 3,3V però non so se questo basterebbe a isolarlo. 

Qui sotto la foto della scheda, magari può servire vederla. Sul retro non ci sono chip, solo le 3 mollettine degli switch a sfioro
![TYWE3S](https://user-images.githubusercontent.com/92910743/138226793-fa4fb774-65e7-450f-8ea8-a986ccf2fe6b.jpg)

Ciao!

**Comment by Scialla** [comment]
a vedere online sembrerebbe che il chip sia flashabile:
https://tasmota.github.io/docs/devices/TYWE3S/

però a vedere su blakadder sembrerebbe che i dispositivi Tuya di ultima generazione montino un FW che previene il flash:
https://templates.blakadder.com/jinvoo_SM-SW101-3.html

Non ho esperienze dirette con Jinvoo, io uso i sonoff, ma ho la sensazione che ci potrai fare poco...

Lascio la parola ha chi ha esperienze dirette

**Reply by reiner348** [reply]
Accidenti! Che fosse possibile prevenire il flash non me l'aspettavo, spero proprio di no!

**Reply by Scialla** [reply]
eh in teoria possono farlo per il flash OTA, ma saldando sui PIN si dovrebbe riuscire a flashare ugualmente, ma non ho esperienze con quel chip..
fatti una ricerca online

**Reply by Scialla** [reply]
c'è una issue lunghissima, se hai voglia di leggertela, magari trovi spunti:
https://github.com/arendst/Tasmota/issues/5377

**Comment by reiner348** [comment]
Ho aggiunto un paio di aggiornamenti nel post principale riguardanti l'MCU e al firmware

**Comment by Franco2255** [comment]
Se hai un pc con linux o un raspberry ci sono diverse guide su come fare a cambiare firmware.
**_LINK RIMOSSO_**

**Reply by Franco2255** [reply]
Questa è una poi ce ne sono altre se per caso hai un pc con linux

**Reply by snakuzzo** [reply]
Non si possono postare link esterni. Rimuovo

**Reply by Franco2255** [reply]
scusa pensavo fosse legato a github

**Comment by reiner348** [comment]
Ce l'ho fatta! Il problema era l'MCU. Ho tagliato il piedino RX che collega l'MCU alla board (cerchiato nella foto), così da isolarlo e il flash è andato al primo colpo!
Ora il problema è che devo ricollegare il piedino, "intelligentemente" è lo stesso pin utilizzato dallo switch centrale...
![image](https://user-images.githubusercontent.com/92910743/138321908-0e0bff6e-9d6a-4485-8808-b8f30d26f2e6.png)

Ora devo capire come far capire a tasmota che l'ho flashato su un chip con 3 input touch, 3 led annessi, led blu e 3 uscite relè, senza avere idea di come sono collegati... Ma questa è un'altra storia!

**Reply by Scialla** [reply]
Ottimo!! Riuscirai sicuramente anche nel resto!

**Comment by TermsOfService22** [comment]
Alla fine sei riuscito a far funzionare il tutto compresi tasti e Led?

**Reply by reiner348** [reply]
Sì sono riuscito! Trai template di tasmota c'è "Jinvoo SM-SW101-3", ho usato quello e funzionano. Ho modificato solo il GPIO del led blu (GPIO00) da led1 a relay4, così posso controllarlo come una uscita separata. In questo modo l'ho potuto usare come luce notturna, ho fatto una logica che li accende tutti (ho 18 jinvoo in casa) alle 20 e li spegne alle 8.

Non si può invece svincolare il led bianco dal relè di uscita corrispondente, quindi se il relè è chiuso il led è acceso e viceversa. E' una logica hardwired nell'interruttore... Però nel mio caso non fa differenza per come ho impostato l'impianto di casa
